fix: OPTIC-1916: Filter Tag is Not Working on Label Studio

luarmr opened the pull request:
The style .lsf-choice_hidden contain no styles this was introduce in a recent migration [here](https://github.com/HumanSignal/label-studio/pull/6187/files#diff-346bb02c58e7899bdebe8ea92303f9fd29f75ba3121f61fb48aa1543c24f70acL17)

The current problem is that the filter does not hide the options. But the right className is being attach

Also fixes: https://github.com/HumanSignal/label-studio/issues/7347

luarmr committed:
fix: OPTIC-1916: Filter Tag is Not Working on Label Studio

bmartel commented on web/libs/editor/src/tags/control/Choice/Choice.scss:
```diff
@@ -18,8 +18,8 @@
 
-  &_leaf &_hidden,
-  &_notLeaf &_hidden {
+  &_leaf&_hidden,
+  &_notLeaf&_hidden {
```
What are the full output classes here?

luarmr replied:
`&=.lsf-choice`

luarmr replied:
well more than that. Notice I try to pair with the original stylus code.

bmartel commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14868734865)

luarmr committed:
change selector scss valid format

luarmr commented on web/libs/editor/src/tags/control/Choice/Choice.scss:
```diff
@@ -18,8 +18,8 @@
-  &_leaf &_hidden,
-  &_notLeaf &_hidden {
+  @at-root #{selector-append(&,_leaf,&,_hidden)},
+  #{selector-append(&,_notLeaf,&,_hidden)} {
```
remove the space showed:
`"&" may only used at the beginning of a compound selector.`
The usual solution `#{&}` did not work, so to avoid using important or attaching extra specificity in front and generating the same selector as in the past and not introduce random bugs, I had to be creative.

bmartel replied:
Tested it locally, works well! For cases like this I have often just used the full selector, but because this component is still using BEM, it also has weird nesting requirements that typically are not a problem to just use the proper class reference in full.

luarmr commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14873008813)

luarmr committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14873008813
